tests/ui/redundant_as_str.rs

```diff
@@ -0,0 +1,6 @@
+#![warn(clippy::redundant_as_str)]
+
+fn main() {
+    let _redundant = "Hello, world!".to_owned().as_str().as_bytes();
```

Comment:
nit: please add tests for other methods that are not redundant to ensure the lint doesn't trigger in those cases

Reply:
I've added a few more test cases that don't trigger the lint.